components/aws/sagemaker/common/_utils.py

```diff
@@ -113,17 +118,38 @@ def print_logs_for_job(cw_client, log_grp, job_name):
         logging.error(e)
 
 
-def get_sagemaker_client(region, endpoint_url=None):
+def get_boto3_session(region, role_arn=None):
+    """Creates a boto3 session, optionally assuming a role"""
+
+    # By default return a basic session
+    if role_arn is None:
+        return boto3
+
+    client = boto3.client("sts", region_name=region,
+                          endpoint_url='https://sts.' + region + '.amazonaws.com')
+    response = client.assume_role(RoleArn=role_arn, RoleSessionName=ASSUME_SESSION_NAME_FORMAT)
```

Comment:
Customer can have long running jobs which can go on for days. 

With oidc/secrets we do not have to worry about refreshing credentials,, assume role will return temporary credentials(max duration - 12 hours which is also controlled by customer), how does the implementation handle such timeouts?

Reply:
That's a fair question. The current implementation only seems to access the client once and then re-uses that client for all of the create and wait calls. I would presume that a long-running job would eventually become Unauthorized to describe the job and subsequently fail.